stix_shifter_modules/nozomi_vantage/stix_transmission/connector.py

```diff
@@ -0,0 +1,272 @@
+from stix_shifter_utils.modules.base.stix_transmission.base_json_sync_connector import BaseJsonSyncConnector
+from stix_shifter_utils.utils.error_response import ErrorResponder
+from stix_shifter_utils.utils import logger
+from .api_client import APIClient
+import json
+
+
+class InvalidMetadataException(Exception):
+    pass
+
+
+class Connector(BaseJsonSyncConnector):
+    NOZOMI_MAX_PAGE_SIZE = 1000
+
+    def __init__(self, connection, configuration):
+        self.api_client = APIClient(connection, configuration)
+        self.logger = logger.set_logger(__name__)
+        self.connector = __name__.split('.')[1]
+        self.expired_token_count = 0
+
+    async def create_results_connection(self, query, offset, length, metadata=None):
+        """
+        Fetching the results using query, offset and length
+        :param query: str, Data Source query
+        :param offset: str, Offset value
+        :param length: str, Length value
+        :param metadata: dict
+        :return: return_obj, dict
+        """
+        data = []
+        try:
+            result_count, total_records, page_number, page_size = self.handle_metadata_and_page_size(offset, length,
```

Comment:
If you use the page_size parameter, there is no need to carry it over through meta data. It'll be a param.

Reply:
Removed page_size in metadata.